### DISCUSSION TITLE
Setting role=combobox shouldn't assume role of the floating element to listbox

### DISCUSSION THREAD

**Discussion by hornta** [floating-ui/floating-ui#2750] [React] [answered]
For this component we use `useRole(context, { role: "combobox" })` which gives the marked element 1 the role of `combobox`, but element 2(the floating element) gets the role of `listbox` which I think is incorrect because we have an inline text field to filter the options and what I read in the spec, the listbox role must only contain elements of role option. Shouldn't the role of element 2 be `dialog` instead?

In my case I should put role="listbox" on element 3, the actual list of options.

I think it should be something like this: https://ariakit.org/examples/select-combobox

![image](https://github.com/floating-ui/floating-ui/assets/4553604/30940904-5351-40ab-ad9a-c4c9e42a224d)

**Comment by atomiks** [comment, marked as answer]
It's not really possible to tell what type of components you're going to build in some cases. 

`combobox` is intended for just a simple input+listbox, since there's no (easy?) way to account for an inner search field which is the actual combobox... unless we add another role type. 

In this case you should use `dialog` role and manually add the roles for now.

**Comment by zhu-hong** [comment]
any way to achieve this feature？
